rfd/0093-offline-access.md

```diff
@@ -0,0 +1,216 @@
+on cluster admins when performing an Auth upgrade. On top of the potential downtime during the upgrade, any issues 
+found as a result of the upgrade require a rollback, which can further extend downtime and lack of connectivity.
+
+## Details
+
+### Auth Connection Required 
+
+The following cluster settings or operations **must** have a connection to Auth server to establish a connection to a node:
+
+#### Label based tsh ssh
+
+```bash
+tsh ssh user@"foo=bar"
+```
+
+For the above command to work, `tsh` needs to be able to resolve all matching nodes which have the label
+`foo=bar`. To determine which nodes match the provided labels `tsh` issues a `proto.AuthService/ListResources`
+request to the Auth server which returns the matching node set. Without a connection to Auth node resolution
+is just not possible.
+
+
+#### Sync Recording modes
+
+Both `proxy-sync` and `node-sync` recording modes require a persistent connection to Auth to upload session recordings. As a result, 
+even if `tsh ssh` doesn't require a connection to Auth, the session cannot be created without being able to start a connection
+for the recordings. This will prevent the connection to the node from being established successfully.
+
+#### Per-Session MFA
+
+When enabled at either the cluster or role level per-session MFA requires completing the MFA ceremony with the 
+Auth server. This means that it is not possible to connect to a resource without `tsh` being able to connect to
+the Auth server to perform the required MFA ceremony.
+
+**Future Improvements**
+There is a scenario in the future that would allow for such behavior to exist. Per 
+[RFD80](https://github.com/gravitational/teleport/blob/master/rfd/0080-hardware-key-support.md#per-session-mfa-configuration), setting `require_session_mfa` to `hardware_key_touch` does not require the Auth server for the MFA ceremony. Instead the PIV stored keys only require a PIV-touch to satisfy the per-session MFA requirement.
+
+#### Strict Recording mode
+
+```yaml
+kind: role
+version: v5
+metadata:
+  name: ssh-strict
+spec:
+  options:
+    record_session:
+      ssh: strict
+```
+
+Strict mode, which is only available when recording at the node, will result in the session failing to launch if the node is unable to record the session.
+
+
+#### Moderated Sessions 
+
+Tracking participants for a moderated session requires persisting the `types.SessionTracker` resource in the backend so that the joining criteria may be evaluated when a user creates or joins a session. In the event that persisting the `types.SessionTracker` fails, the session must be aborted to prevent any unauthorized access. This is true for both proxy and node recording mode. 
+
+#### Web UI
+
+All access via the Web UI requires a connection to Auth. None of the auth connector resources are replicated to the Proxy which means users are unable to login to the Web UI without
```

Comment:
Hypothetical: could the Web UI, assuming everything else is already loaded, launch just the SSH session shell without Auth? How much effort do we think that would be?

Reply:
Thanks for the detailed explanation. Keeping whatever info we need to connect directly from the Servers page makes sense to me, but I haven't looked closely at it. (Also there's the ticket you mentioned.)